tensorboard/backend/BUILD

```diff
@@ -66,13 +66,7 @@ py_library(
         "//tensorboard:errors",
         "//tensorboard:plugin_util",
         "//tensorboard/backend/event_processing:data_provider",
```

Comment:
Need to remove this dep to actually prune the build path, no?

```
$ git status
HEAD detached at nf/refactor-application
nothing to commit, working tree clean
$ git rev-parse @
799f17f214d66e5d526374f8d1c79b50b3cc3a0f
$ bazel query 'somepath(//tensorboard/backend:application, //tensorboard/compat:tensorflow)'
//tensorboard/backend:application
//tensorboard/backend/event_processing:data_provider
//tensorboard/backend/event_processing:event_accumulator
//tensorboard/compat:tensorflow
```

Unfortunately, doing so breaks some tests:

```
//tensorboard/plugins/image:images_plugin_test
//tensorboard/plugins/image:images_plugin_notf_test
//tensorboard/plugins/histogram:histograms_plugin_test
//tensorboard/plugins/histogram:histograms_plugin_notf_test
```

…because apparently I have a blind spot for missing this dep; sorry.
(Would send a PR to fix, but I think it might be logistically easier for
you to do so?)

Reply:
Ugh, yes, thanks. Not sure where I lost this change since obviously that's the point of all this...

Re: the deps - sure, sent you #3654 to fix. You know, I thought when fixing some of the other plugins in #3641 that I should check all of the plugins just in case, and I spot-checked to see that they had the event multiplexer dep, but didn't consider that they were missing the data provider dep :/